src/client/session/cluster_time.rs

```diff
@@ -0,0 +1,36 @@
+use bson::{Document, TimeStamp};
+use serde::{Deserialize, Serialize};
+
+/// Struct modeling a cluster time reported by the server.
+///
+/// See [the MongoDB documentation](https://docs.mongodb.com/manual/core/read-isolation-consistency-recency/)
+/// for more information.
+#[derive(Debug, Deserialize, Clone, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct ClusterTime {
+    cluster_time: TimeStamp,
+    signature: Document,
+}
+
+impl std::cmp::PartialEq for ClusterTime {
+    fn eq(&self, other: &ClusterTime) -> bool {
+        self.cluster_time == other.cluster_time
+    }
+}
+
+impl std::cmp::Eq for ClusterTime {}
```

Comment:
Out of curiosity, does it compile if you move the `Eq` from `derivative` to `derive`? My thinking is that it will, but I haven't run into this exact thing before, so I'm not sure

Reply:
nope, doesn't compile. And yeah, same issue as before.

Reply:
Interesting, good to know